File: queue.py
# Queue implementation in Python
class Queue():
    def __init__(self, max_size):
        self.queue = []
        self.max_size = max_size

    def enqueue(self, val):
        if not self.IsFull():
            self.queue.append(val)
        else:
            print("Queue is full")

    def dequeue(self):
        holder = self.queue[0]
        del self.queue[0]
        return holder

    def peek(self):
        return self.queue[0]

    def rear(self):
        return self.queue[-1]

    def IsFull(self):
        return len(self.queue) == self.max_size

    def IsEmpty(self):
        return len(self.queue) == 0

File: Queue/queue.js
class Queue {
    constructor(max_size) {
        this.queue = [];
        this.max_size = max_size;
    };

    enqueue(val) {
        if (!this.isFull()) {
            this.queue.push(val);
        }
        else {
            console.log("Queue is Full");
        }
    };

    dequeue() {
        val = this.queue[0];
        this.queue.splice(0, 1);
        return val;
    };

    peek() {
        return this.queue[0];
    };

    rear(){
        return this.queue[this.queue.length - 1];
    };

    isFull() {
        return this.queue.length == this.max_size;
    };

    isEmpty() {
        return this.queue.length == 0;
    };
};
